common/src/main/java/org/keycloak/common/util/IoUtils.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.common.util;
+
+import java.io.Console;
+
+public class IoUtils {
+
+    public static String readFromConsole(String kind, String defaultValue) {
+        Console cons = System.console();
+        if (cons == null) {
+            if (defaultValue != null) {
+                return defaultValue;
+            }
+            throw new RuntimeException(String.format("Console is not active, but %s is required", kind));
+        }
+        char[] passwd;
+        String prompt = String.format("Enter %s", kind) + (defaultValue != null ? String.format(" [%s]:", defaultValue) : ":");
+        if ((passwd = cons.readPassword(prompt)) != null) {
```

Comment:
I think the old method name(`readSecret`) was more suitable as it represented it reads sensitive info, which is hidden. I'd consider `readFromConsole` to also see what I'm typing. For instance, I think the `client-id` might be visible to users when they're typing it, right?

At least rename it to `readSecretFromConsole` ?

Reply:
I'll update it to take a "password" parameter, which will address your comment below.